android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java

```diff
@@ -406,6 +411,35 @@ public void clearNotifications() {
         notificationManager.cancelAll();
     }
 
+    public void clearDeliveredNotifications(ReadableArray identifiers) {
+      NotificationManager notificationManager = notificationManager();
+      for (int index = 0; index < identifiers.size(); index++) {
+        String id = identifiers.getString(index);
+        Log.i(LOG_TAG, "Removing notification with id " + id);
+        notificationManager.cancel(Integer.parseInt(id));
```

Comment:
One gotcha, if there is a "tag" set on the notification, this will no longer.

You have to call a different [cancel](https://developer.android.com/reference/android/app/NotificationManager.html#cancel(java.lang.String, int))

In trying to stay consistent with the iOS removeDeliveredNotifications interface (which takes a list of identifiers) we lose that capability.

It seems that was an existing limitation with the similar functionality in the cancelLocalNotifications interface -
 https://github.com/zo0r/react-native-push-notification/issues/568

Reply:
Can you call `toArrayList` on the ReadableArray and access the elements directly?

https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/bridge/ReadableArray.java

`
for (String id: identifiers.toArrayList()) {
  ...
}
`

Reply:
Don't think that will work.  toArrayList returns an ArrayList<Object>

I'd have to do a cast in the loop:
```
for (Object id: identifiers.toArrayList()) {
    notificationManager.cancel(Integer.parseInt((String)id));
}
```
Not sure that reads better